### ISSUE TITLE
Error:  non-matching or non-convertible constant type for const initializer

### ISSUE THREAD

**Issue by jamesdolan** [KhronosGroup/glslang#2016]
After updating to latest glslang master I am getting the following error on some rather trivial shader code in global scope. `defaultVersion=420`...

`ERROR: common.glsl:7: 'const' : non-matching or non-convertible constant type for const initializer`

```
const float kPI = 3.1415926535897932384626433832795;
const float k2PI = kPI * 2.0; // <-- error here
```

**Comment by jamesdolan** [comment]
FYI `90cecb52` was the last version I tested that worked.

**Comment by jamesdolan** [comment]
Did a binary search, this is the change that breaks it https://github.com/KhronosGroup/glslang/commit/3c0d2e5b671c667abdcf7863ab6d08a8dde7e456

**Comment by jamesdolan** [comment]
Turning `ENABLE_HLSL` does workaround the problem

**Comment by johnkslang** [comment]
How are you compiling?  I don't get this error with a simple 460 vertex shader, but don't know what input/language/version/options you are compiling with.

**Comment by johnkslang** [comment]
> Did a binary search, this is the change that breaks it 3c0d2e5

So that commit should have **fixed** HLSL on/off issues by making headers more stable.

Is it possible you did not completely rebuild everything relying on headers when doing the binary search?  An out of date .o would explain an random-like error like this.

(And such problems are reduced now, with that commit, rather than being increased.)

**Comment by jamesdolan** [comment]
Definitely wasn't a stale build issue, repros reliably even now if I go back and clean my entire repo.

Preprocessor settings... 'ENABLE_OPT=1', 'AMD_EXTENSIONS', 'NV_EXTENSIONS'

GLSL compilation looks something like...
```
    glslang::TShader shader(EShLangVertex);
    shader.setStringsWithLengthsAndNames(... );
    shader.setEntryPoint("main");
    shader.setSourceEntryPoint("main");
    if (!shader.parse(&glslang::DefaultTBuiltInResource, 420, ECoreProfile, true, true, messages)) {
```

Are you testing with `ENABLE_HLSL` and `ENABLE_GLSLANG_WEB` off? Because default appears to be on.

**Comment by johnkslang** [comment]
Now I think it is the uninitialized member found by @stuartcarnie in #2020, with fix going in now as #2021.
